app/main.py

```diff
@@ -1,3 +1,62 @@
-def shop_trip():
-    # write your code here
-    pass
+from decimal import Decimal, ROUND_HALF_UP
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+with open("app/config.json", "r") as json_file:
+    data_dict = json.load(json_file)
+
+FUEL_PRICE = Decimal(data_dict["FUEL_PRICE"])
+
+customers = [Customer(*data_dict["customers"][i].values())
+             for i in range(len(data_dict["customers"]))]
+
+shops = [Shop(*data_dict["shops"][i].values())
+         for i in range(len(data_dict["shops"]))]
+
+
+def shop_trip() -> None:
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+
+        min_total_cost = Decimal("Infinity")
+        min_shop = None
+
+        for shop in shops:
+            fuel_cost = customer.calc_trip_cost(shop, FUEL_PRICE) * 2
+            products_price = shop.calc_total_cost(customer.product_cart)
+            total_cost = fuel_cost + products_price
+            total_cost = total_cost.quantize(Decimal("0.01"),
+                                             rounding=ROUND_HALF_UP)
+
+            print(f"{customer.name}'s trip to "
+                  f"the {shop.name} costs {total_cost}")
+
+            if total_cost <= min_total_cost:
+                min_total_cost = total_cost
+                min_shop = shop
```

Comment:
I would suggest to create separate function to calculate total cost as well (directly in main or in Customer class), but overall structure looks better!